homeassistant/helpers/entity.py

```diff
@@ -58,7 +58,11 @@
 FLOAT_PRECISION = abs(int(math.floor(math.log10(abs(sys.float_info.epsilon))))) - 1
 
 
-ENTITY_CATEGORIES_SCHEMA: Final = vol.In(ENTITY_CATEGORIES)
+# pylint: disable-next=invalid-name
+def ENTITY_CATEGORIES_SCHEMA(value: Any | None) -> EntityCategory:
+    """Validate entity category configuration."""
+    value = vol.In(ENTITY_CATEGORIES)(value)
+    return EntityCategory(value)
```

Comment:
```suggestion
ENTITY_CATEGORIES_SCHEMA: Final = vol.Coerce(EntityCategory)
```
wouldn't that do the same?

Reply:
It works, but the error message is not helpful.

With this PR:
```
voluptuous.error.MultipleInvalid: value must be one of ['config', 'diagnostic', 'system'] for dictionary value @ data['entity_category']
```

With the suggestion:
```
voluptuous.error.MultipleInvalid: expected EntityCategory for dictionary value @ data['entity_category']
```

Reply:
ah right, this is indeed helpful. Should it be adopted to something like `cv.enum_value` - could be used eg. for `components.number.NumberMode`

Reply:
I agree. Want to open a PR @farmio?

Reply:
Maybe we could update `Coerce` upstream to print nice error messages 🤔

Reply:
Yes I can probably do that in the next days. 
Not sure about `Coerce` though. A type may not be iterable (`Coerce(int)`) at all or have too many members to print in a readable way etc.